core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverManager.java

```diff
@@ -196,10 +264,35 @@ default SolverJob<Solution_, ProblemId_> solve(ProblemId_ problemId,
      *        If null it defaults to logging the exception as an error.
      * @return never null
      */
+    default SolverJob<Solution_, ProblemId_> solve(ProblemId_ problemId,
+                                           Function<? super ProblemId_, ? extends Solution_> problemFinder,
+                                           Consumer<? super Solution_> finalBestSolutionConsumer,
+                                           BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler) {
+        return builder(problemId, problemFinder)
+                .withFinalBestSolutionConsumer(finalBestSolutionConsumer)
+                .withExceptionHandler(exceptionHandler)
+                .solve();
+    }
+
+    /**
+     * As defined by {@link #solve(Object, Function, Consumer, BiConsumer)}.
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     *        {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, function that returns a {@link PlanningSolution}, usually with uninitialized planning
+     *        variables
+     * @param finalBestSolutionConsumer sometimes null, called only once, at the end, on a consumer thread
+     * @param exceptionHandler sometimes null, called if an exception or error occurs.
+     *        If null it defaults to logging the exception as an error.
+     * @param configOverride sometimes null, includes settings that override the default configuration
+     * @return never null
+     */
     SolverJob<Solution_, ProblemId_> solve(ProblemId_ problemId,
-            Function<? super ProblemId_, ? extends Solution_> problemFinder,
-            Consumer<? super Solution_> finalBestSolutionConsumer,
-            BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler);
+                                           Function<? super ProblemId_, ? extends Solution_> problemFinder,
+                                           Consumer<? super Solution_> finalBestSolutionConsumer,
+                                           BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler,
+                                           SolverConfigOverride configOverride);
```

Comment:
Leave the config override business to the builder.

Reply:
Let's wait for the design meeting to define the next steps here.

Reply:
The new design includes only two methods:

1. `<Solution_, ProblemId_> buildSolver();`
2. `<Solution_, ProblemId_> solve(SolverJobConfig<Solution_, ProblemId_> solverJobConfig);`

Reply:
I don't think that's correct.
As far as I remember, we agreed there would be:

- solve(...) with the most basic arguments.
- solveAndListen(...) with the most basic arguments.
- solveBuilder().

Everything else would be deprecated.

Reply:
Upon reviewing the code, I noticed that the method `solve(...)` already includes the `solveAndListen(...)` logic. The only necessary addition would be best consumer validation. Do you still think we should add the method `<Solution_, ProblemId_> solveAndListen(SolverJobConfig<Solution_, ProblemId_> solverJobConfig);`?

Reply:
No. We `SolverJobConfig` should not be the argument of any method in the public API.